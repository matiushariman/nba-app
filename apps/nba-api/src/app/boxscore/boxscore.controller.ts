import { Controller, Get, Param } from '@nestjs/common';

import { BoxscoreService } from './boxscore.service';

import type { GetBoxscoreByGameIdReq } from '@nba-app/types-nba';

@Controller('boxscore')
export class BoxscoreController {
  constructor(private readonly boxscoreService: BoxscoreService) {}

  @Get(':gameId')
  getBoxscoreByGameId(@Param() params: GetBoxscoreByGameIdReq) {
    return this.boxscoreService.getBoxscoreByGameId(params.gameId);
  }
}
