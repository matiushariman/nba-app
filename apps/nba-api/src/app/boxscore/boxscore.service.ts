import { Injectable } from '@nestjs/common';
import { HttpService } from '@nestjs/axios';
import { catchError, firstValueFrom } from 'rxjs';

import type {
  GetBoxscoreByGameIdRes,
  GetBoxscoreByGameIdReq,
} from '@nba-app/types-nba';

@Injectable()
export class BoxscoreService {
  constructor(private readonly httpService: HttpService) {}

  async getBoxscoreByGameId(
    gameId: GetBoxscoreByGameIdReq['gameId']
  ): Promise<GetBoxscoreByGameIdRes> {
    const { data } = await firstValueFrom(
      this.httpService
        .get<GetBoxscoreByGameIdRes>(
          `https://cdn.nba.com/static/json/liveData/boxscore/boxscore_${gameId}.json`
        )
        .pipe(
          catchError(() => {
            throw 'Oops';
          })
        )
    );

    return data;
  }
}
