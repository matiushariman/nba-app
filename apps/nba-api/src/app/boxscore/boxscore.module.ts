import { Module } from '@nestjs/common';
import { HttpModule } from '@nestjs/axios';

import { BoxscoreService } from './boxscore.service';
import { BoxscoreController } from './boxscore.controller';

@Module({
  imports: [HttpModule],
  providers: [BoxscoreService],
  controllers: [BoxscoreController],
})
export class BoxscoreModule {}
