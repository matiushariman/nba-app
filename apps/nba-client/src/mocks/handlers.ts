import { boxscoreHandlers } from './boxscore/handlers';
import { gamesHandlers } from './games/handlers';

export const handlers = [...boxscoreHandlers, ...gamesHandlers];
