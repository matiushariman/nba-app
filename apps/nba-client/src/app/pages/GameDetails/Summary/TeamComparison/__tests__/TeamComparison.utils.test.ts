import {
  getPtsOffTurnoversStats,
  getOffReboundsStats,
  getDefReboundsStats,
  getFastBreakPtsStats,
  getBlocksStats,
  getStealsStats,
  getReboundsStats,
  getAssistsStats,
  getFieldGoalStats,
  getThreePtStats,
  getFreeThrowStats,
  getPercentage,
} from '../TeamComparison.utils';
import { mockBoxcore } from '../../../../../../mocks/boxscore/handlers';

const { homeTeam, awayTeam } = mockBoxcore[0].game;

describe('pages/GameDetails/Summary/TeamComparison/TeamComparison.utils', () => {
  describe('getPtsOffTurnoversStats', () => {
    it('should return object that matches snapshot', () => {
      expect(getPtsOffTurnoversStats(awayTeam, homeTeam)).toMatchSnapshot();
    });
  });

  describe('getOffReboundsStats', () => {
    it('should return object that matches snapshot', () => {
      expect(getOffReboundsStats(awayTeam, homeTeam)).toMatchSnapshot();
    });
  });

  describe('getDefReboundsStats', () => {
    it('should return object that matches snapshot', () => {
      expect(getDefReboundsStats(awayTeam, homeTeam)).toMatchSnapshot();
    });
  });

  describe('getFastBreakPtsStats', () => {
    it('should return object that matches snapshot', () => {
      expect(getFastBreakPtsStats(awayTeam, homeTeam)).toMatchSnapshot();
    });
  });

  describe('getBlocksStats', () => {
    it('should return object that matches snapshot', () => {
      expect(getBlocksStats(awayTeam, homeTeam)).toMatchSnapshot();
    });
  });

  describe('getStealsStats', () => {
    it('should return object that matches snapshot', () => {
      expect(getStealsStats(awayTeam, homeTeam)).toMatchSnapshot();
    });
  });

  describe('getReboundsStats', () => {
    it('should return object that matches snapshot', () => {
      expect(getReboundsStats(awayTeam, homeTeam)).toMatchSnapshot();
    });
  });

  describe('getAssistsStats', () => {
    it('should return object that matches snapshot', () => {
      expect(getAssistsStats(awayTeam, homeTeam)).toMatchSnapshot();
    });
  });

  describe('getFieldGoalStats', () => {
    it('should return object that matches snapshot', () => {
      expect(getFieldGoalStats(awayTeam, homeTeam)).toMatchSnapshot();
    });
  });

  describe('getThreePtStats', () => {
    it('should return object that matches snapshot', () => {
      expect(getThreePtStats(awayTeam, homeTeam)).toMatchSnapshot();
    });
  });

  describe('getFreeThrowStats', () => {
    it('should return object that matches snapshot', () => {
      expect(getFreeThrowStats(awayTeam, homeTeam)).toMatchSnapshot();
    });
  });

  describe('getPercentage', () => {
    it('should return number in percentage', () => {
      expect(getPercentage(0.01)).toEqual(1);
    });
  });
});
