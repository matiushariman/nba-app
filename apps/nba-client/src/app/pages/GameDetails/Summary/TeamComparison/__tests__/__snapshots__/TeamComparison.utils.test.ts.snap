// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/GameDetails/Summary/TeamComparison/TeamComparison.utils getAssistsStats should return object that matches snapshot 1`] = `
Object {
  "awayTeam": Object {
    "percentage": 100,
    "primaryText": 34,
    "teamTricode": "PHX",
  },
  "homeTeam": Object {
    "percentage": 91.17647058823529,
    "primaryText": 31,
    "teamTricode": "NOP",
  },
  "label": "Assists",
}
`;

exports[`pages/GameDetails/Summary/TeamComparison/TeamComparison.utils getBlocksStats should return object that matches snapshot 1`] = `
Object {
  "awayTeam": Object {
    "percentage": 100,
    "primaryText": 5,
    "teamTricode": "PHX",
  },
  "homeTeam": Object {
    "percentage": 60,
    "primaryText": 3,
    "teamTricode": "NOP",
  },
  "label": "Blocks",
}
`;

exports[`pages/GameDetails/Summary/TeamComparison/TeamComparison.utils getDefReboundsStats should return object that matches snapshot 1`] = `
Object {
  "awayTeam": Object {
    "percentage": 63.888888888888886,
    "primaryText": 23,
    "teamTricode": "PHX",
  },
  "homeTeam": Object {
    "percentage": 100,
    "primaryText": 36,
    "teamTricode": "NOP",
  },
  "label": "Defensive Rebounds",
}
`;

exports[`pages/GameDetails/Summary/TeamComparison/TeamComparison.utils getFastBreakPtsStats should return object that matches snapshot 1`] = `
Object {
  "awayTeam": Object {
    "percentage": 62.5,
    "primaryText": 5,
    "teamTricode": "PHX",
  },
  "homeTeam": Object {
    "percentage": 100,
    "primaryText": 8,
    "teamTricode": "NOP",
  },
  "label": "Fast Break Points",
}
`;

exports[`pages/GameDetails/Summary/TeamComparison/TeamComparison.utils getFieldGoalStats should return object that matches snapshot 1`] = `
Object {
  "awayTeam": Object {
    "percentage": 46.6666666666667,
    "primaryText": "49/105",
    "secondaryText": "46.7%",
    "teamTricode": "PHX",
  },
  "homeTeam": Object {
    "percentage": 58.1395348837209,
    "primaryText": "50/86",
    "secondaryText": "58.1%",
    "teamTricode": "NOP",
  },
  "label": "Field Goals",
}
`;

exports[`pages/GameDetails/Summary/TeamComparison/TeamComparison.utils getFreeThrowStats should return object that matches snapshot 1`] = `
Object {
  "awayTeam": Object {
    "percentage": 76.4705882352941,
    "primaryText": "13/17",
    "secondaryText": "76.5%",
    "teamTricode": "PHX",
  },
  "homeTeam": Object {
    "percentage": 75,
    "primaryText": "21/28",
    "secondaryText": "75.0%",
    "teamTricode": "NOP",
  },
  "label": "Free Throws",
}
`;

exports[`pages/GameDetails/Summary/TeamComparison/TeamComparison.utils getOffReboundsStats should return object that matches snapshot 1`] = `
Object {
  "awayTeam": Object {
    "percentage": 100,
    "primaryText": 19,
    "teamTricode": "PHX",
  },
  "homeTeam": Object {
    "percentage": 36.84210526315789,
    "primaryText": 7,
    "teamTricode": "NOP",
  },
  "label": "Offensive Rebounds",
}
`;

exports[`pages/GameDetails/Summary/TeamComparison/TeamComparison.utils getPtsOffTurnoversStats should return object that matches snapshot 1`] = `
Object {
  "awayTeam": Object {
    "percentage": 92.5925925925926,
    "primaryText": 25,
    "teamTricode": "PHX",
  },
  "homeTeam": Object {
    "percentage": 100,
    "primaryText": 27,
    "teamTricode": "NOP",
  },
  "label": "Points Off Turnovers",
}
`;

exports[`pages/GameDetails/Summary/TeamComparison/TeamComparison.utils getReboundsStats should return object that matches snapshot 1`] = `
Object {
  "awayTeam": Object {
    "percentage": 97.67441860465115,
    "primaryText": 42,
    "teamTricode": "PHX",
  },
  "homeTeam": Object {
    "percentage": 100,
    "primaryText": 43,
    "teamTricode": "NOP",
  },
  "label": "Total Rebounds",
}
`;

exports[`pages/GameDetails/Summary/TeamComparison/TeamComparison.utils getStealsStats should return object that matches snapshot 1`] = `
Object {
  "awayTeam": Object {
    "percentage": 70,
    "primaryText": 7,
    "teamTricode": "PHX",
  },
  "homeTeam": Object {
    "percentage": 100,
    "primaryText": 10,
    "teamTricode": "NOP",
  },
  "label": "Steals",
}
`;

exports[`pages/GameDetails/Summary/TeamComparison/TeamComparison.utils getThreePtStats should return object that matches snapshot 1`] = `
Object {
  "awayTeam": Object {
    "percentage": 34.2105263157895,
    "primaryText": "13/38",
    "secondaryText": "34.2%",
    "teamTricode": "PHX",
  },
  "homeTeam": Object {
    "percentage": 32,
    "primaryText": "8/25",
    "secondaryText": "32.0%",
    "teamTricode": "NOP",
  },
  "label": "3-Pointers",
}
`;
